cpp/test/sg/tsne_test.cu

```diff
@@ -128,6 +129,13 @@ class TSNETest : public ::testing::TestWithParam<TSNEInput> {
     // Allocate memory
     rmm::device_uvector<float> X_d(n * p, stream);
     raft::update_device(X_d.data(), dataset.data(), n * p, stream);
+
+    rmm::device_uvector<float> Xtranspose(n * p, stream);
+
+    raft::update_device(Xtranspose.data(), X_d.data(), n * p, stream);
```

Comment:
Use `raft::copy_async` for this since it is a device -> device transfer. The underlying code doesn't change it is just for readability.